### DISCUSSION TITLE
Should simplex noise be added? Patent expired Jan 8, 2022

### DISCUSSION THREAD

**Discussion by TrickMe** [godotengine/godot-proposals#5007] [General Discussions]
Should simplex noise be added?
Apparently the [patent](https://patents.google.com/patent/US6867776) has expired on Jan 8, 2022. [wiki](https://en.wikipedia.org/wiki/Simplex_noise#Legal_status)
I haven't used noise yet and don't know about possible benefits of simplex noise over openSimplex noise, but I want to spread the word.
Discuss.

**Comment by theraot** [comment]
Be aware of https://github.com/godotengine/godot/pull/56718

**Comment by TokisanGames** [comment]
Noise was already overhauled and merged into gd4. It now supports your own modules, so you can use one that includes simplex in your own projects. There won't be any other library embedded in the engine now that FastNoiseLite has made it in to gd4. 

Zylann and I both have FastNoise wrappers, and I have a FastNoiseSIMD wrapper; modules for gd3 with simplex. 

Artistically, simplex can be replicated by perlin or opensimplex so it's unnecessary.
